### PR TITLE
CON-3027 Changes for authentication policy

### PR DESCRIPTION
**Pull request by Howesy77**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Extensions/IdentityExtensions.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 namespace SFA.DAS.ProviderCommitments.Web.Extensions
 {
-    public static class IIdentityExtensions
+    public static class IdentityExtensions
     {
         public static string Upn(this IIdentity identity)
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Extensions/IdentityExtensions.cs:10]: if the class is renamed, shouldn't the argument type also be renamed?

Comment (Howesy77) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Extensions/IdentityExtensions.cs]: Reverted

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Authorization/ProviderHandler.cs [hunk 2]
```diff
@@ -27,24 +29,46 @@ protected override Task HandleRequirementAsync(AuthorizationHandlerContext conte
...
                 }
             }
-            context.Succeed(requirement);
+
             return Task.CompletedTask;
         }
+
+        private bool HasServiceAuthorization(AuthorizationHandlerContext context)
+        {
+            static bool ServiceClaimFinderPredicate(Claim c)
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Authorization/ProviderHandler.cs:59]: for the sake of a 1 line lambda, I don't think the local function is necessary.

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Extensions/ServiceClaimExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        [TestCase(ServiceClaim.DAB, new ServiceClaim[] { ServiceClaim.DAV }, false)]
+        [TestCase(ServiceClaim.DAC, new ServiceClaim[] { ServiceClaim.DAV }, false)]
+        [TestCase(ServiceClaim.DAV, new ServiceClaim[] { ServiceClaim.DAV }, true)]
+        [TestCase(ServiceClaim.DAA, new ServiceClaim[] { }, false)]
+        [TestCase(ServiceClaim.DAB, new ServiceClaim[] { }, false)]
+        [TestCase(ServiceClaim.DAC, new ServiceClaim[] { }, false)]
+        [TestCase(ServiceClaim.DAV, new ServiceClaim[] { }, false)]
+        public void ShouldReturnHasPermission(ServiceClaim minimumServiceClaim, ServiceClaim[] actualServiceClaims, bool expected)
+        {
+            var claims = actualServiceClaims.Select(a => new Claim(ProviderClaims.Service, a.ToString()));
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Extensions/ServiceClaimExtensionsTests.cs:58]: can we add some none das claims to this list such as DCFT to cover the user having Service claims but none being valid for our service?

**Review by cofaulco — COMMENTED** [review]
Couple of comments

**Review by cofaulco — APPROVED** [review]

**Pull request merged**